pandas/core/tools/datetimes.py

```diff
@@ -302,6 +302,9 @@ def _convert_listlike_datetimes(arg, box, format, name=None, tz=None,
         return result
 
     except ValueError as e:
+        if "unless utc=True" not in str(e):
+            # raised by array_to_datetime
+            raise e
```

Comment:
This condition doesn't cause any test failures, but it isn't 100% obvious what this branch is supposed to catch.  If we can confirm that this is supposed to catch only ValueErrors raised by `array_to_datetime`, then the try/except here can be simplified quite a bit.

Reply:
what do you need this for? I really really don't like things like this.

Reply:
The issue is that there is too much inside the "try" part of this try/except so it isn't clear what this is supposed to catch.  If I'm right that we only really want to catch this one error raised by `tslib.array_to_datetime`, then the other stuff can be moved outside of the try/except and this check becomes unnecessary.

Reply:
that may be true (or not), but its orthogonal to this PR. let's remove

Reply:
OK, will open a separate PR for this step